src/app/task.service.ts

```diff
@@ -24,12 +25,12 @@ export class TaskService {
   }
 
   getTasksInProgress(): Observable<Task[]> {
-    this.tasks = this.tasks.filter(t => t.status !== 1);
-    return of(this.tasks);
+    const taskInProgress = this.tasks.filter(t => t.status !== Status.DONE);
+    return of(taskInProgress);
   }
 
   finishTask(task: Task): Observable<Task> {
-    task.status = 1;
+    task.status = Status.DONE;
```

Comment:
Is the finish working? You should find this task in the tasks list and then change it status to DONE.